libpod/events/events.go

```diff
@@ -0,0 +1,268 @@
+	// Create ...
+	Create Status = "create"
+	// Exec ...
+	Exec Status = "exec"
+	// Export ...
+	Export Status = "export"
+	// History ...
+	History Status = "history"
+	// Import ...
+	Import Status = "import"
+	// Init ...
+	Init Status = "init"
+	// Inspect ...
+	Inspect Status = "inspect"
+	// Kill ...
+	Kill Status = "kill"
+	// LoadFromArchive ...
+	LoadFromArchive Status = "status"
+	// Mount ...
+	Mount Status = "mount"
+	// Pause ...
+	Pause Status = "pause"
+	// Prune ...
+	Prune Status = "prune"
+	// Pull ...
+	Pull Status = "pull"
+	// Push ...
+	Push Status = "push"
+	// Refresh ...
+	Refresh Status = "refresh"
+	// Remove ...
+	Remove Status = "remove"
+	// Restart ...
+	Restart Status = "restart"
+	// Save ...
+	Save Status = "save"
+	// Start ...
+	Start Status = "start"
+	// StartAndAttach ...
+	StartAndAttach Status = "startandattach"
+	// Stop ...
+	Stop Status = "stop"
+	// Sync ...
+	Sync Status = "sync"
+	// Tag ...
+	Tag Status = "tag"
+	// Unmount ...
+	Unmount Status = "unmount"
+	// Unpause ...
+	Unpause Status = "unpause"
+	// Untag ...
+	Untag Status = "untag"
+	// Wait ...
+	Wait Status = "wait"
+)
+
+// EventFilter for filtering events
+type EventFilter func(*Event) bool
+
+// NewEvent creates a event struct and populates with
+// the given status and time.
+func NewEvent(status Status) Event {
+	return Event{
+		Status: status,
+		Time:   time.Now(),
+	}
+}
+
+// Write will record the event to the given path
+func (e *Event) Write(path string) error {
+	f, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0700)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	if _, err := f.WriteString(e.ToJSONString()); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Recycle checks if the event log has reach a limit and if so
+// renames the current log and starts a new one.  The remove bool
+// indicates the old log file should be deleted.
+func (e *Event) Recycle(path string, remove bool) error {
+	return errors.New("not implemented")
+}
+
+// ToJSONString returns the event as a json'ified string
+func (e *Event) ToJSONString() string {
+	b, _ := json.Marshal(e)
+	return fmt.Sprintf("%s\n", b)
+}
+
+// ToHumanReadable returns human readable event as a formatted string
+func (e *Event) ToHumanReadable() string {
+	var humanFormat string
+
+	switch e.Type {
+	case Container, Volume, Pod:
+		humanFormat = fmt.Sprintf("%s %s %s %s (image=%s, name=%s)", e.Time, e.Type, e.Status, e.ID, e.Image, e.Name)
+	case Image:
+		humanFormat = fmt.Sprintf("%s %s %s %s %s", e.Time, e.Type, e.Status, e.ID, e.Name)
+	}
+	return humanFormat
+}
+
+// NewEventFromString takes stringified json and converts
+// it to an event
+func NewEventFromString(event string) (*Event, error) {
+	e := Event{}
+	if err := json.Unmarshal([]byte(event), &e); err != nil {
+		return nil, err
+	}
+	return &e, nil
+
+}
+
+// ToString converts a Type to a string
+func (t Type) ToString() string {
+	return fmt.Sprintf("%s", t)
```

Comment:
Can you return t?  Or string(t)

Reply:
string(t) ought to be fine.

Also rename to `String()` - it's a generally-used interface in Golang that a lot of methods accept